app/routers/meds.py

```diff
@@ -0,0 +1,364 @@
+                        amount (int) - Daily dosage.
+                        early (time) - Earliest reminder time.
+                        late (time) - Latest reminder time.
+                        minimum (time) - Minimal interval between reminders.
+                        maximum (time) - Maximal interval between reminders.
+                        note (str) - Additional description.
+                        start (datetime) - First reminder date and time.
+                        end (datetime) - Last reminder date and time.
+    """
+    name = form['name']
+    start_date = get_time_from_string(form['start'])
+    first = get_time_from_string(form['first'])
+    end_date = get_time_from_string(form['end'])
+    amount = int(form['amount'])
+    early = get_time_from_string(form['early'])
+    late = get_time_from_string(form['late'])
+    minimum = get_time_from_string(form['min'])
+    maximum = get_time_from_string(form['max'])
+    first_time = first if first else early
+    start = datetime.combine(start_date, first_time)
+    if late <= early:
+        end_date += timedelta(1)
+    end = datetime.combine(end_date, late)
+    note = form['note']
+    return (
+        name, first, amount, early, late, minimum, maximum, note, start, end)
+
+
+def convert_time_to_minutes(time_obj: time) -> int:
+    """Returns time object as minutes.
+
+    Args:
+        time_obj (time): Time object to convert to minutes.
+
+    Returns:
+        int: Total minutes in time object.
+    """
+    return round(time_obj.hour * 60 + time_obj.minute)
+
+
+def get_interval_in_minutes(early: time, late: time) -> int:
+    """Returns interval between 2 time objects in minutes.
+
+    Args:
+        early (time): Earlier time object.
+        late (time): Later time object. Interpreted as following day if earlier
+        than early.
+
+    Returns:
+        int: Interval between time objects in minutes.
+    """
+    if early == late:
+        0
+    extra = 1 if early > late else 0
+    early_date = datetime.combine(datetime.min, early)
+    late_date = datetime.combine(datetime.min + timedelta(extra), late)
+    interval = late_date - early_date
+    return round(interval.seconds / 60)
+
+
+def validate_amount(amount: int, minimum: time, early: time,
+                    late: time) -> bool:
+    """Returns True if interval is sufficient for reminder amount with minimal
+    interval constraint, False otherwise
+
+    Args:
+        amount (int): Reminder amount.
+        minimum (time): Minimal interval between reminders.
+        early (time) - Earliest reminder time.
+        late (time) - Latest reminder time.
+
+    Returns:
+        bool: True if interval is sufficient for reminder amount with minimal
+              interval constraint, False otherwise
+    """
+    min_time = (amount - 1) * convert_time_to_minutes(minimum)
+    interval = get_interval_in_minutes(early, late)
+    if min_time <= interval:
+        return True
+    return False
```

Comment:
return min_time <= interval

Reply:
Sometimes we try to make the most simple things appear more difficult than they actually are.
Sometimes we're just missing silly mistakes :D